services/gps2tsdb/docker-compose.yml

```diff
@@ -0,0 +1,18 @@
+version: '3.1'
+services:
+  # docker run -it -v /var/run/dbus:/var/run/dbus --rm --name my-running-app my-python-app
+  gps2tsdb:
+    build:
+      # Cannot use `./` due to this issue https://github.com/docker/compose/issues/3530#issuecomment-222490501
+      context: ../gps2tsdb
+    restart: always
+    environment:
+      db_user: "avena"
+      db_host: "postgres"
+      db_database: "avena"
+      db_password: "password"
+      db_port: "5432"
+    depends_on:
+      - postgres
```

Comment:
Maybe I'm not explaining what I assumed the docker-compose would be like. Not saying it should be this way, but maybe it will clarify earlier comments. That said, I can sort of see automated tests around something like this.

```suggestion
version: '3.1'
services:
  # docker-compose run --rm gps2tsdb bash run-tests.sh
  gps2tsdb:
    build:
      context: ./
    restart: never
    environment:
      db_user: "avena"
      db_host: "postgres"
      db_database: "avena"
      db_password: "password"
      db_port: "5432"
    depends_on:
      - tsdb
      - gpsd
    volumes:
      - dbus:/var/run/dbus
  tsdb:
    image: timescale/timescaledb:latest-pg12
    restart: never
    environment:
      POSTGRES_PASSWORD: "password"
      POSTGRES_USER: "avena"
      POSTGRES_DB: "avena"
      TIMESCALEDB_TELEMETRY: "off"
    ports:
      - 5432:5432
    volumes:
      - postgres:/var/lib/postgresql/data
      - ./postgres/schema:/docker-entrypoint-initdb.d
      - dbus:/var/run/dbus
  gpsd:
    image: <...some gpsd image ...>
    restart: never
    volumes:
      - dbus:/var/run/dbus
  gpsfake:
    image: <...some gpsfake image ...>
    restart: never
    depends_on:
      - gpsd
  sd-bus:
    image: <... some dbus-deamon image ...>
    restart: never
    volumes:
      - dbus:/var/run/dbus
volumes:
  postgres:
  dbus:
```

Reply:
Lets discuss this in the 7/29 meeting... I agree with you on what the `docker-compose.yml` file should be used for in the future, however I think we should not change it until we write the automated test. The way it is right now allows me to manually test everything and until we have an automated test ready to replace that lets not take that ability away

Reply:
A bit more clarification based on your comment here: https://github.com/OATS-Group/isoblue-avena/pull/45#discussion_r461653429

Right now the `docker-compose.yml` files are setup so I can do a `docker-compose -f xxx/docker-compose.yml -f yyy/docker-compose.yml ... up` to bring up the services I want for manual testing or deployment. In the (very very near) future we should transition to ansible/k3s/k8s for deployment and `docker-compose.yml` for automated testing

That is how I see it at least. Lets discuss in the 7/29 meeting

Reply:
If you remove the parts where we don't yet have an image made for it, it still works just like your current implementation except the `gps2tsdb` service is in charge of the postgres/timescale version, the schema, etc. and a developer just has to run `docker-compose --rm star gps2tsdb` to get the test images spun up. 

I don't think that's far from what we talked about how Avena may work. `gps2tsdb` would call on the timescale controller/manger to create it a user, database and it would establish the schema within that namespace.